packages/test/src/test-isolation.ts

```diff
@@ -118,3 +142,46 @@ test('Shared global state is frozen', withReusableContext, async (t) => {
   )) as WorkflowFailedError;
   t.is(err.cause?.message, 'Cannot add property a, object is not extensible');
 });
+
+export async function sharedGlobalReassignment(): Promise<[string, string, string]> {
+  type ConsoleExtended = Console & { wfid: string };
+  // Replace the `console` global by a new object
+  globalThis.console = { ...globalThis.console, wfid: workflowInfo().workflowId } as ConsoleExtended;
+  const middle = (globalThis.console as ConsoleExtended).wfid;
+  await sleep(50);
+  return [workflowInfo().workflowId, middle, (globalThis.console as ConsoleExtended).wfid];
+}
+
+test('Reassign shared global state', withReusableContext, async (t) => {
+  const { createWorker, taskQueue, env } = t.context;
+  const worker = await createWorker();
+  await worker.runUntil(async () => {
+    const [res1, res2] = await Promise.all([
+      env.client.workflow.execute(sharedGlobalReassignment, { taskQueue, workflowId: randomUUID() }),
+      env.client.workflow.execute(sharedGlobalReassignment, { taskQueue, workflowId: randomUUID() }),
```

Comment:
This may be racy in CI and may not actually provide the test coverage you want. There's nothing guaranteeing that the workflows will run concurrently.